cpp/src/arrow/adapters/orc/adapter_util.h

```diff
@@ -34,8 +34,21 @@ namespace orc {
 
 Status GetArrowType(const liborc::Type* type, std::shared_ptr<DataType>* out);
 
+Status GetORCType(const DataType& type, ORC_UNIQUE_PTR<liborc::Type>* out);
+
+Status GetORCType(const Schema& schema, ORC_UNIQUE_PTR<liborc::Type>* out);
+
 Status AppendBatch(const liborc::Type* type, liborc::ColumnVectorBatch* batch,
                    int64_t offset, int64_t length, ArrayBuilder* builder);
+
+Status FillBatch(const DataType* type, liborc::ColumnVectorBatch* cbatch,
+                 int64_t& arrowOffset, int64_t& orcOffset, int64_t length, Array* parray,
+                 std::vector<bool>* incomingMask = NULLPTR);
+
+Status FillBatch(const DataType* type, liborc::ColumnVectorBatch* cbatch,
+                 int64_t& arrowIndexOffset, int& arrowChunkOffset, int64_t length,
+                 ChunkedArray* pchunkedArray);
```

Comment:
I have renamed all the `FillBatch` `WriteBatch`. As for `AppendBatch` they currently remain since `ReadBatch` already exists and has a different usage.